libkbfs/tlf_journal.go

```diff
@@ -532,25 +558,48 @@ func (j *tlfJournal) doBackgroundWork(ctx context.Context) <-chan error {
 	go func() {
 		defer j.wg.Done()
 		errCh <- j.flush(ctx)
+		close(errCh)
 	}()
 	return errCh
 }
 
-// We don't guarantee that pause/resume requests will be processed in
-// strict FIFO order. In particular, multiple pause requests are
-// collapsed into one (also multiple resume requests), so it's
-// possible that a pause-resume-pause sequence will be processed as
-// pause-resume. But that's okay, since these are just for infrequent
-// ad-hoc testing.
+// We don't guarantee that background pause/resume requests will be
+// processed in strict FIFO order. In particular, multiple pause
+// requests are collapsed into one (also multiple resume requests), so
+// it's possible that a pause-resume-pause sequence will be processed
+// as pause-resume. But that's okay, since these are just for
+// infrequent ad-hoc testing.
 
-func (j *tlfJournal) pauseBackgroundWork() {
+func (j *tlfJournal) pause(pauseType tlfJournalPauseType) {
+	j.pausedLock.Lock()
+	defer j.pausedLock.Unlock()
+	j.pausedType |= pauseType
+
+	if j.pausedType > pauseType {
```

Comment:
Multiple calls to pause with the same pause type will still signal, right? Shouldn't those be filtered out too (by checking if `j.pausedType == 0` before setting it)? Or if it's harmless, may as well not do the check at all.